Fixed bug where editor wouldn't launch

StandingPadAnimations opened the pull request:
For some reason, maybe a quirk of my terminal or a Go quirk, Discordo couldn't launch an editor. Launching Neovim would hang and Nano would stright up fail. Some poking around found that it was due to Stdout. For some reason, setting Stdout to a string just didn't work.

Now I could try to capture Stdout, but I decided to opt for the simple route of "create a temporary file and extract the message from there".

This is my first time touching Go at all, so it's not the best, but it fixes this issue.

StandingPadAnimations committed:
Fixed bug where editor wouldn't launch
For some reason, maybe a quirk of my terminal or a Go quirk, Discordo
couldn't launch an editor. Launching Neovim would hang and Nano would
stright up fail. Some poking around found that it was due to Stdout. For
some reason, setting Stdout to a string just didn't work.

Now I could try to capture Stdout, but I decided to opt for the simple
route of "create a temporary file and extract the message from there".

This is my first time touching Go at all, so it's not the best, but it
fixes this issue.

sewnie commented on message_input.go:
```diff
@@ -116,10 +116,26 @@ func (mi *MessageInput) launchEditorAction() {
+	// Stdout to a variable actually causes editors to not
+	// work for some reason, so we're going with the more
+	// reliable method.
+	msg_file := os.TempDir() + "/discord_msg.txt"
```
```suggestion
	msg_file := filepath.Join(os.TempDir(), "discord_msg-" + os.Getpid() + ".txt")
```

ayn2op replied:
https://pkg.go.dev/os#CreateTemp

StandingPadAnimations committed:
Changed how the temp file is made and deleted
To prevent conflict issues, the temporary file made now is
`discord_msg-PID.txt` instead of `discord_msg.txt`, to prevent issues
with multiple instances. In addition, the file is now deleted whenever
an error occurs, even if it hasn't been written to, to prevent multiple
of these from existing.

StandingPadAnimations committed:
Fixed missing module and adding an int to a string
filepath.Join is in the filepath module, which wasn't imported, so it
wouldn't compile. os.Getpid returns an int, which also caused a compile
time error.

sewnie reviewed (commented):
Why not defer the `os.Remove(msg_file)` call instead of calling it twice?

StandingPadAnimations committed:
File removing is now deferred
By defering the remove call, the temporary file will now be deleted when
the function returns, regardless of failiure or not.

StandingPadAnimations committed:
Replaced the old creation method with CreateTemp
As suggested, I've replaced file creation with CreateTemp, which
drastically cleans up the code.

StandingPadAnimations committed:
Fixed typo in comment

Co-authored-by: jumps are op <jumpsareop@gmail.com>

ayn2op commented on message_input.go:
```diff
@@ -116,10 +116,28 @@ func (mi *MessageInput) launchEditorAction() {
+	// capturing Stdout to a variable actually causes editors 
+	// to not work for some reason, so we're going with the more
+	// reliable method.
+	f, err := os.CreateTemp("", "discord_msg")
```
```suggestion
	f, err := os.CreateTemp("", config.Name + "-*.md")
```

StandingPadAnimations committed:
Temp files now created as markdown files

Co-authored-by: ayn2op <119342035+ayn2op@users.noreply.github.com>

ayn2op committed:
Merge branch 'main' into main

ayn2op committed:
Merge branch 'main' into main

StandingPadAnimations committed:
Merge remote-tracking branch 'upstream/main'

ayn2op committed:
Merge branch 'main' into main

ayn2op commented on cmd/run/message_input.go:
```diff
@@ -111,10 +111,28 @@ func (mi *MessageInput) launchEditorAction() {
+	// reliable method.
+	f, err := os.CreateTemp("", config.Name + "-*.md")
+	if err != nil {
+		log.Println(err)
```
```suggestion
		log.Println(err)
		return
```

ayn2op commented on cmd/run/message_input.go:
```diff
@@ -111,10 +111,28 @@ func (mi *MessageInput) launchEditorAction() {
+	f.Close()
+	
+	// Defer this so the file is deleted when the
+	// function returns, regardless of failiure or not
```
Typo.

```suggestion
	// function returns, regardless of failure or not
```

ayn2op commented on cmd/run/message_input.go:
```diff
@@ -123,6 +141,15 @@ func (mi *MessageInput) launchEditorAction() {
+	// One may ask "Why don't we remove the file earlier?". Well,
+	// the file won't contain any message up until this point, and the file
+	// is created in /tmp anyway so it'll be cleared on a reboot.
+	var msg, read_err = os.ReadFile(f.Name())
```
Use `:=` instead of `var`.

```suggestion
	msg, read_err := os.ReadFile(f.Name())
```

StandingPadAnimations committed:
Added return statement as suggested

Co-authored-by: ayn2op <119342035+ayn2op@users.noreply.github.com>

StandingPadAnimations committed:
Fixed typo in comment

Co-authored-by: ayn2op <119342035+ayn2op@users.noreply.github.com>

StandingPadAnimations committed:
Swapped var for :=

Co-authored-by: ayn2op <119342035+ayn2op@users.noreply.github.com>

ayn2op commented on cmd/run/message_input.go:
```diff
@@ -123,6 +142,15 @@ func (mi *MessageInput) launchEditorAction() {
+	// One may ask "Why don't we remove the file earlier?". Well,
+	// the file won't contain any message up until this point, and the file
+	// is created in /tmp anyway so it'll be cleared on a reboot.
+	msg, read_err := os.ReadFile(f.Name())
```
Convert `read_err` to camel case.

StandingPadAnimations committed:
Changed read_err to readErr
